In [14]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

# Q1.Cutlets

In [3]:
data=pd.read_csv('Cutlets.csv')
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [4]:
Unit_A=pd.Series(data.iloc[:,0])
Unit_A

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [5]:
Unit_B=pd.Series(data.iloc[:,1])
Unit_B

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [6]:
p_value=stats.ttest_ind(Unit_A,Unit_B)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

The p value is 0.47 i.e 0.47 > 0.05. Therefore we do not reject Null Hypothesis.



# Q2. LabTAT

In [7]:
data=pd.read_csv('LabTAT.csv')
data.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [8]:
p_value=stats.f_oneway(data.iloc[:,0],data.iloc[:,1],data.iloc[:,2],data.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [10]:
p_value[1]

2.1156708949992414e-57

The p value is 2.1156708949992414e-57 i.e 2.1156708949992414e-57 < 0.05. Therefore we reject Null Hypothesis.



# Q3. BuyerRatio

In [11]:
data=pd.read_csv('BuyerRatio.csv')
data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [12]:
ob=np.array([[50,142,131,70],[435,1523,1356,750]])
ob

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [15]:
chi2_contingency(ob)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

The p value is 0.6603094907091882 i.e 0.6603094907091882 > 0.05. Therefore we do not reject Null Hypothesis.



# Q4. CustomerOrderForm

In [16]:
data=pd.read_csv('Costomer+OrderForm.csv')
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [17]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [18]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [19]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [20]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [21]:
obv=np.array([[271,267,269,280],[29,33,31,20]])
obv

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [22]:
chi2_contingency(obv)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

The p value is 0.2771020991233135 i.e 0.2771020991233135 > 0.05. Therefore we do not reject Null Hypothesis.